client/src/components/MemberList.js

```diff
@@ -8,46 +9,65 @@ import TableHead from '@material-ui/core/TableHead';
 import TableRow from '@material-ui/core/TableRow';
 import Paper from '@material-ui/core/Paper';
 
+var rows = [];
+
 const useStyles = makeStyles({
     table: {
         minWidth: 650,
     },
 });
 
-function createMemberData(id, username, gender, age) {
-    return { id, username, gender, age };
+//Info stored: username, gender*, age*, status*, budget*
+function createMemberData(id, username, gender, age, status, budget) {
+    return { id, username, gender, age, status, budget };
 }
 
-var rows = [];
-
 export default function MemberList(props) {
-    //window.alert(props.data);
-    if(props.data !== undefined) {
-        rows = props.data;
+    const [state, setState] = useState(0);
+    // if (state > 0) {
+    //     isClicked();
+    //     //rows = props.data;
+    // }
+    // if (state == 0) {
+    //     isClicked2();
+    //     //rows = props.data;
+    // }
+    if (props.data !== undefined) {
+        if(rows !== props.data) {
+            rows = props.data;
+            setState(0);
+        }
+        
     }
```

Comment:
Hooks are pretty cool and awesome. But, for this component, there's actually no need to store a local state. We can just use the props directly in our table JSX below.

In fact, the current code as written is actually a bit of an anti-pattern (bad practice) in React coding, [see here for more info](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#anti-pattern-unconditionally-copying-props-to-state).

Reply:
I added it as I might have done the states improperly before, leading to the table to not update when the props were updated. The upcoming update should fix that issue.